src/altinn-app-frontend/src/features/form/containers/Form.tsx

```diff
@@ -161,14 +149,44 @@ export function Form() {
         spacing={3}
         alignItems='flex-start'
       >
-        {currentView === currentLayout &&
-          filteredLayout &&
-          filteredLayout.map((component) => {
-            return renderLayoutComponent(component, layout);
-          })}
+        {mainComponents.map((component) =>
+          renderLayoutComponent(component, layout),
+        )}
+        <ErrorReport components={errorReportComponents} />
       </Grid>
-    </div>
+    </>
   );
 }
 
-export default Form;
+function topLevelComponents(layout: ILayout) {
+  const inGroup = new Set<string>();
+  layout.forEach((component) => {
+    if (component.type === 'Group') {
+      const childList = component.edit?.multiPage
+        ? component.children.map((childId) => childId.split(':')[1] || childId)
+        : component.children;
+      childList.forEach((childId) => inGroup.add(childId));
+    }
+  });
+  return layout.filter((component) => !inGroup.has(component.id));
+}
+
+function extractBottomButtons(layout: ILayout) {
+  const extract = new Set<ComponentTypes>([
+    'NavigationButtons',
+    'Button',
+    'PrintButton',
+  ]);
+
+  const toMainLayout: ILayout = [];
+  const toErrorReport: ILayout = [];
+  for (const component of [...layout].reverse()) {
```

Comment:
Was the intent to break before scanning the whole layout?

Reply:
If I understood you correctly; yes. Earlier code iterated the layout from the bottom-up to extract the bottom-most of these components (`NavigationButtons`, `Button` and `PrintButton`) and break the iteration when it finds other components. This is basically the same, but gives you two outputs (one for the regular layout, another with these components split into its own layout so they can be injected into `ErrorReport` as children).